lib/mongo/grid/fs_bucket.rb

```diff
@@ -487,8 +487,26 @@ def files_name
 
       def ensure_indexes!
         if files_collection.find({}, limit: 1, projection: { _id: 1 }).first.nil?
-          chunks_collection.indexes.create_one(FSBucket::CHUNKS_INDEX, :unique => true)
-          files_collection.indexes.create_one(FSBucket::FILES_INDEX)
+          create_index_if_missing!(files_name, FSBucket::FILES_INDEX)
+        end
+
+        if chunks_collection.find({}, limit: 1, projection: { _id: 1 }).first.nil?
+          create_index_if_missing!(chunks_name, FSBucket::CHUNKS_INDEX, :unique => true)
```

Comment:
This seems to create each index based on the presence of data in the respective backing collection. I had a different understanding of the mechanism so the docs should be updated to match the actual behavior.

Reply:
I'm going based on the [spec's "Before write operations"](https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#id19) guidance:

```
Immediately before the first write operation on an instance of a GridFSBucket class is attempted (and not earlier), drivers MUST:

determine if the files collection is empty using the primary read preference mode.
and if so, create the indexes described above if they do not already exist
To determine whether the files collection is empty drivers SHOULD execute the equivalent of the following shell command:

> db.fs.files.findOne({}, { _id : 1 })
If no document is returned the files collection is empty.
```

Reply:
I haven't addressed this yet, so may still have to revise the docs before this can be merged